doc/python-type-hinting.md

```diff
@@ -64,24 +62,100 @@ def test(self, var: Union[None, int, str]) -> None:
         return [value + 1 for value in values]
 ```
 
-## Forward declarations
+## Self Type
+
+To annotate methods that return an instance of their class, the `Self` type is used as it is bound to it's encapsulating class. In the example below, the type checker will correctly infer the type of `Circle().set_scale(0.5)` to be `Circle`:
+
+```python
+from typing import Self
+
+class Shape:
+    def set_scale(self, scale: float) -> Self:
+        self.scale = scale
+        return self
+
+
+class Circle(Shape):
+    def set_radius(self, radius: float) -> Self:
+        self.radius = radius
+        return self
+```
+
+This is recommended as forward declarations are now redundant in 3.10
+https://peps.python.org/pep-0673/
+
+## Type Alias
+
+Use the special annotation `TypeAlias` to declare type aliases more explicitly so type checkers are able to distinguish between type aliases and ordinary assignments:
+
+```python
+MyType: TypeAlias = "ClassName"
+def foo() -> MyType: ...
+```
+
+## Type Ignore
+
+To mark portions of the program that should not be covered by type hinting, use the following on a particular line:
+
+```python
+# type: ignore
+```
+
+A `# type: ignore` comment on a line by itself at the top of a file silences all errors in the file.
+
+`# type: ignore` should only be used when unavoidable. Ensure that a comment is added to explain why it has been used and have a prefix of `Type ignore:`.
+
```

Comment:
I would add an example here to make this clearer, maybe the comment and a line below with the #type: ignore added?

Reply:
Example added to better illustrate the use of `Type ignore`